smart-contracts/contracts/UnlockTestV3.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity 0.4.24;
+
+/**
+ * @title Example upgrade of the Unlock contract
+ * @author Julien Genestoux (unlock-protocol.com)
+ * @author HardlyDifficult
+ * 
+ * This is a copy paste of the current Unlock contract, with comments removed.
+ * Then a few small changes have been made, higlighted with comments.
+ */
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "./Lock.sol";
+
+
+contract UnlockTestV3 is Ownable {
+
+  struct LockBalances {
+    bool deployed;
+    uint totalSales; 
+    uint yieldedDiscountTokens;
+  }
+
+  modifier onlyFromDeployedLock() {
+    require(locks[msg.sender].deployed, "Only from previously deployed locks");
+    _;
+  }
+
+  uint public grossNetworkProduct;
+
+  uint public totalDiscountGranted;
+
+  mapping (address => LockBalances) public locks;
+
+  event NewLock(
+    address indexed lockOwner,
+    address indexed newLockAddress
+  );
+
+  bool internal initialized;
+
+  // Example new data (which must come after the original data)
+  bool internal initializedV3;
```

Comment:
**MUST** ?

Reply:
Yes. See [Preserving the storage structure](https://docs.zeppelinos.org/docs/advanced.html)